client/src/components/overview/ImageNavigator.jsx

```diff
@@ -1,27 +1,163 @@
-      </div>
-      <div class="image-navigator-item">
-        <div class="icon"></div>
-        <div></div>
-      </div>
-      <div class="image-navigator-item">
-        <div class="thumbnail"></div>
-        <div></div>
+class ImageNavigator extends React.Component {
+  constructor(props = {
+    thumbnails,
+    useIcons,
+    length,
+    selectedId,
+    onClick
+  }) {
+    super(props);
+
+    this.injectProperties = this.injectProperties.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.handleDecrement = this.handleDecrement.bind(this);
+    this.handleIncrement = this.handleIncrement.bind(this);
+
+    this.model = new ImageNavigatorModel(this.injectProperties(true));
+    this.onClick = this.props.onClick ? this.props.onClick : () => {};
+
+    this.state = {
+      startIndex: this.model.startIndex
+    };
+  }
+
+  injectProperties(initialize) {
+    return {
+      thumbnails: this.props.thumbnails,
+      useIcons: this.props.useIcons ? this.props.useIcons : false,
+      length: this.props.length,
+      maxItems: this.props.maxItems ? this.props.maxItems : 7,
+      startIndex: initialize ? 0 : this.state.startIndex
+    };
+  }
+
+  handleClick(e) {
+    if (e.currentTarget.hasAttribute('data-image-id')) {
+      e.stopPropagation();
+      let imageId = parseInt(e.currentTarget.dataset.imageId);
+      if (imageId !== this.props.selectedId) {
+        this.onClick(imageId);
+      }
+    }
+  }
+
+  handleDecrement(e) {
+    e.stopPropagation();
+    if (this.model.decrementIndex()) {
+      this.setState({
+        startIndex: this.model.startIndex
+      });
+    }
+  }
+
+  handleIncrement(e) {
+    e.stopPropagation();
+    if (this.model.incrementIndex()) {
+      this.setState({
+        startIndex: this.model.startIndex
+      });
+    }
+  }
+
+  render() {
+    console.log('Rendering image navigator');
+    this.model = new ImageNavigatorModel(this.injectProperties());
+    const model = this.model;
+
+    const { startIndex, endIndex } = model;
+    const useCase = model.getUseCase();
+    const items = model.getItemUrls();
+
+    const itemClass = model.getItemParentClass();
+    const itemChildClass = model.getItemChildClass();
+    const itemChildClassSelected = itemChildClass + ' selected';
+
+    let itemKey = startIndex;
+
+    return (
+      <div class="image-navigator">
+        {
+          model.firstIndexNotVisible(startIndex) &&
+          <div class={itemClass} onClick={this.handleDecrement}>
+            <div class="navigator-align">
+              <div class="navigator decrement">
+                <svg viewBox="0,0,1000,837">
+                  <path
+                    d="M71.9816+837.371L500.008+309.729L928.018+837.371L1000+778.965L500.008+162.629L0+778.965L71.9816+837.371Z"
+                  />
+                </svg>
+              </div>
+            </div>
+          </div>
+        }
+        {
+          items.map(item => (
+            <div key={itemKey++}
```

Comment:
Are you sure these aren't going to be reordered? Would using `item.id` wreck something? I haven't followed your logic far enough to know.